tree: remove pointer definition for DBox2D

otan opened the pull request:
Box2D should never actually be nil inside, so remove that requirement.

Resolves #53013.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53134)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 15 of 15 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53134#-:-MFCGopJBMpnhrgyoZnK:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

otan committed:
tree: remove pointer definition for DBox2D

Box2D should never actually be nil inside, so remove that requirement.

Release note: None

otan commented:
bors r=sumeerbhola
